pkg/cmd/taskrun/delete.go

```diff
@@ -180,20 +196,42 @@ func taskRunLister(p cli.Params, keep int, kind string, cs *cli.Clients) func(st
 		if kind == "Task" {
 			trs.Items = taskpkg.FilterByRef(trs.Items, string(v1beta1.NamespacedTaskKind))
 		}
-		return keepTaskRuns(trs, keep), nil
+		todelete, _ := keepTaskRunsByNumber(trs, keep)
+		return todelete, nil
 	}
 }
 
-func allTaskRunNames(cs *cli.Clients, keep int, ns string) ([]string, error) {
+func allTaskRunNames(cs *cli.Clients, keep, since int, ns string) ([]string, []string, error) {
+	var todelete, tokeep []string
+
 	taskRuns, err := trlist.TaskRuns(cs, metav1.ListOptions{}, ns)
 	if err != nil {
-		return nil, err
+		return todelete, tokeep, err
+	}
+
+	if since > 0 {
```

Comment:
cant we do like

```
if since > 0 {
		return keepTaskRunsByAge(taskRuns, since), nil
}
return keepTaskRunsByNumber(taskRuns, keep), nil

Reply:
This is cleaner but the method is supposed to return 3 values.

Reply:
Adding an extra nil to the return value of `keepTaskRunsByNumber` and `keepTaskRunsByAge` is doable though.